File: Programmers/Level2/is_pair(p)/is_pair.py
def is_pair(s):
    ret = 0
    first = True
    check = ""
    for c in s:
        if c=='(' or c==')':
            check = c
            if first:
                first = False
                if c==')':
                    return False
            if c=='(':
                ret+=1
            else:
                ret-=1
    if check==')' and ret == 0:
        return True
    return False

# 아래는 테스트로 출력해 보기 위한 코드입니다.
print( is_pair("(hello)()"))
print( is_pair(")("))
print( is_pair("(()))("))

File: Programmers/Level2/is_pair(p)/is_pair.js
function is_pair(s){
	var open = 0;
	for(var i=0; i<s.length; ++i)
	{
		if(s[i]=="(")
		{
			++open;
		}
		else if(s[i]==")")
		{
			--open;
		}
		if(open<0)
		{
			return false;
		}
	}
  return open==0 ? true:false;
}

// 아래는 테스트로 출력해 보기 위한 코드입니다.
console.log( is_pair("(hello)()") )
console.log( is_pair(")(") )